File: C++/프로그래머스/1/12901. 2016년/2016년.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int a, int b) {
    vector<int> monthdate = {0,0,31,29,31,30,31,30,31,31,30,31,30};
    vector<string> day = {"FRI","SAT", "SUN","MON","TUE","WED", "THU"};
    
    int before = 0;
    for(auto& m : monthdate) {
        m += before;
        before = m;
    }
    return day[(monthdate[a] + b - 1) % 7];
}

File: Python3/프로그래머스/1/12901. 2016년/2016년.py
def solution(a, b):
    monthdate = [0,0,31,29,31,30,31,30,31,31,30,31,30]
    for i in range(1, len(monthdate)):
        monthdate[i] += monthdate[i - 1];
    day = ["FRI","SAT","SUN","MON","TUE","WED","THU"]
    
    return day[(monthdate[a] + b - 1) % 7]